src/dsp_tools/utils/shared.py

```diff
@@ -86,9 +86,13 @@ def try_network_action(
                 return action(*args, **kwargs)
             else:
                 return action()
+        except (TimeoutError, ReadTimeout):
+            logger.error(f"Exceeded timeout while executing {action_as_str}")
+            raise BaseError(f"Exceeded timeout while executing {action_as_str}") from None
```

Comment:
This is based on the assumption that it is not worth retrying after a timeout. (Other types of errors lead to a retry, and also a timeout error led to a retry until now.)

Reply:
I haven't given this much thought, just now. But is this indeed an assumption that holds? I don't know in which scenarios you would actually get a timeout, but if there is one, where the network/server could recover afterwards, then you might want to retry after all